src/libs/extra/metadata.liq

```diff
@@ -0,0 +1,98 @@
+# Store and retrieve file covers using metadata. This returns a set of
+# getter/setter methods that can be used to store and retrieve cover art.
+# Typical usage is to set cover art in a `on_metadata` handler and retrieve
+# it in a `video.add_image` operator.
+# @category Metadata
+# @param ~default Default cover file when no cover is available
+# @param ~mime_types Recognized mime types and their corresponding file extensions.
+def file.cover.manager(
+  ~id=null(),
+  ~mime_types=[
+    ("image/gif", "gif"),
+    ("image/jpeg", "jpeg"),
+    ("image/png", "png"),
+    ("image/webp", "webp")
+  ],
+  ~default
+) =
+  id = string.id.default(id, default="cover")
+  temp_dir = file.temp_dir("cover-manager")
+  on_cleanup({file.rmdir(temp_dir)})
+
+  current_cover_file = ref(default)
+
+  def set_current_cover(filename) =
+    current_cover = current_cover_file()
+    if
+      current_cover != default
+    then
+      log.info(
+        label=id,
+        "Removing #{string.quote(current_cover)}"
+      )
+      file.remove(current_cover)
+    end
+    current_cover_file := filename
+  end
```

Comment:
I wonder if `current_cover_file` might point to a non-existent file after removing the previous one.

Reply:
Yeah we can inverse the order of operations to avoid a race condition.